connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerInfo.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Connect Worker system and runtime information.
+ */
+public class WorkerInfo {
+    private static final Logger log = LoggerFactory.getLogger(WorkerInfo.class);
+    private static final RuntimeMXBean RUNTIME;
+
+    static {
+        RUNTIME = ManagementFactory.getRuntimeMXBean();
+    }
+
+    private final Map<String, Object> values;
+
+    /**
+     * Constructor.
+     */
+    public WorkerInfo() {
+        this.values = new LinkedHashMap<>();
+        addRuntimeInfo();
+    }
+
+    /**
+     * Log the values of this object at level INFO.
+     */
+    // Equivalent to logAll in AbstractConfig
+    public void logAll() {
+        StringBuilder b = new StringBuilder();
+        b.append(getClass().getSimpleName());
+        b.append(" values: ");
+        b.append(Utils.NL);
+
+        for (Map.Entry<String, Object> entry : new TreeMap<>(values).entrySet()) {
+            b.append('\t');
+            b.append(entry.getKey());
+            b.append(" = ");
+            b.append(format(entry.getValue()));
+            b.append(Utils.NL);
+        }
+        log.info(b.toString());
+    }
+
+    private static Object format(Object value) {
+        return value == null ? "NA" : value;
+    }
+
+    /**
+     * Collect general runtime information.
+     */
+    protected void addRuntimeInfo() {
+        List<String> jvmArgs = RUNTIME.getInputArguments();
```

Comment:
Should we add some more stuff to round this out (and make use of all the support for reporting more than one value...), e.g. some other `RuntimeMXBean` info? For example, classpath info seems like it'd be useful (probably more so before `plugin.path`, but still probably handy from time to time).

Reply:
Honestly I found more interesting info within `MemoryMXBean ` and `OperatingSystemMXBean` and others, but I stumbled on the need to define a `DataUnit` (similar to `TimeUnit`) to format byte sizes and I punted it for the future. 

For classpath in specific I wouldn't mind adding it, but I'm wondering whether it'll create confusion when used besides `plugin.path`. 

Other properties in `RuntimeMXBean` are not that interesting, but I could add some if you want now. Final thoughts on adding classpath?

Reply:
yeah, other stuff from `RuntimeMXBean` was just a suggestion, obviously i wouldn't limit to that. classpath was the main one i saw that is useful.

i don't think that it's too confusing since it is also alongside a bunch of other general system info. also, we still support the classpath approach, we just won't fix conflicts. we've definitely had cases on the mailing list that ended up being classpath issues that we probably could have spotted the likely root cause more quickly if we had had access to the classpath info.

Reply:
More than happy to add it.

Reply:
Is it also worthwhile to add the plugin path here as well?

Reply:
I'm fine either way, the main difference is that everything listed so far is global and not really specific to Connect (and can therefore be processed and logged immediately upon startup) whereas getting `plugin.path` requires doing some argument parsing and would be logged later on by the normal argument parsing `logAll` call.

Reply:
Given that `plugin.path` is already printed as part of `StandaloneConfig` or `DistributedConfig` I'm inclined to punt including this info here (although I agree that it also belongs close to the `jvm.classpath`). A future addition could mean that the code that calls `initInfo.logAll();` should move below config parsing, which probably is fine. 

Let me know what you think.